### PR TITLE
Parse request/response before caching

### PR DESCRIPTION
**Pull request by robiscoding**
# Description

- To prevent the redis cache in the hosted api environment from growing too large, this change parses the information going into the cache if the `TINY_CACHE` env var is set to true
- Updates docker publish script to target correct platform

## Type of change

Please delete options that are not relevant.

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [X] New feature (non-breaking change which adds functionality)
-   [ ] This change requires a documentation update

## How has this change been tested, please provide a testcase or example of how you tested the change?

- Tested in a local redis cache that the parse req/res get written to the cache when the env var is true, and that the full req/res are written when the var is false

## Any specific deployment considerations

For example, documentation changes, usability, usage/costs, secrets, etc.

## Docs

-   [ ] Docs updated? What were the changes:

### CODE REVIEW

File: inference/core/env.py [hunk 1]
```diff
@@ -56,6 +56,11 @@
 # Maximum batch size for GAZE, default is 8
 GAZE_MAX_BATCH_SIZE = int(os.getenv("GAZE_MAX_BATCH_SIZE", 8))
 
+# If true, this will store smaller representations of the inference in the cache
+# This is useful for reducing the size of the cache, but will not allow for the
+# retrieval of the original inference
+CONDENSED_CACHED = str2bool(os.getenv("CONDENSED_CACHED", False))
```
Comment (paulguerrie) [inline inference/core/env.py:62]: Should we default this to True?

File: inference/core/cache/transforms.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+from inference.core.env import CONDENSED_CACHED
+from inference.core.logger import logger
+from inference.core.version import __version__
+
+
+def to_cachable_inference_item(
+    infer_request: InferenceRequest,
+    infer_response: Union[InferenceResponse, list[InferenceResponse]],
+) -> dict:
+    request = jsonable_encoder(infer_request.dict())
```
Comment (paulguerrie) [inline inference/core/cache/transforms.py:17]: I think we can exclude fields which might make this line run faster. So maybe, something like:
```suggestion
    excluded_fields = ['image'] if CONDENSED_CACHE else []
    request = jsonable_encoder(infer_request.dict(exclude=excluded_fields))
```

Docs for `.dict(...)`: https://docs.pydantic.dev/1.10/usage/exporting_models/

**Review by paulguerrie — CHANGES_REQUESTED** [review]
Looking good so far! Just a couple small comments.

**Review by paulguerrie — APPROVED** [review]

**Review by probicheaux — APPROVED** [review]
lgtm

**Review by SolomonLake — APPROVED** [review]

**Pull request merged**